Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php

```diff
@@ -53,7 +53,7 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^dcngettext$' => null,
         '^dgettext$' => null,
         '^dio_.*$' => null,
-        '^dirname$' => null,
+        '^dirname$' => '\Magento\Framework\Filesystem\Io\File::getPathInfo [\'dirname\']',
```

Comment:
How about the following:

```suggestion
        '^dirname$' => '\Magento\Framework\Filesystem\Io\File::getPathInfo($filePath)[\'dirname\']',
```

The same relates to `basename`.